pom.xml

```diff
@@ -208,7 +208,10 @@
         <version.error-prone-orig>2.24.1</version.error-prone-orig>
         <version.error-prone-slf4j>0.1.22</version.error-prone-slf4j>
         <version.guava-beta-checker>1.0</version.guava-beta-checker>
-        <version.jdk>11</version.jdk>
+        <!-- Though the project targets JDK 11, the code base does use more
+        recent language constructs, thanks to the Jabel compiler plugin. -->
+        <version.jdk.source>17</version.jdk.source>
```

Comment:
I see https://github.com/bsideup/jabel/pull/177 is resolved. Why can't we have our source use Java 21? :eyes:

Reply:
Ah, that ticket is about supporting JDK 20 and 21 for building, not for extending support for additional language features. Right now Jabel doesn't unlock support for any JDK 18+ language features, so there's no value in targeting e.g. Java 21.

---

I did have a quick look to see whether adding `PATTERN_SWITCH` [here](https://github.com/bsideup/jabel/blob/ad06a0be7d6f3bb41fdba149df21b4963036fe6c/jabel-javac-plugin/src/main/java/com/github/bsideup/jabel/JabelCompilerPlugin.java#L149-L165) would "just work", but the answer is "no".

Test code:
```diff
$ git stash show -p
diff --git a/example/src/main/java/com/example/JabelExample.java b/example/src/main/java/com/example/JabelExample.java
index 49fc126..d465164 100644
--- a/example/src/main/java/com/example/JabelExample.java
+++ b/example/src/main/java/com/example/JabelExample.java
@@ -23,6 +23,10 @@ public class JabelExample {
             case 2, 3 -> "two or three";
             default -> new JabelExample().new Inner().innerPublic();
         };
+        result = switch ((Object) args[0]) {
+            case String s -> s;
+            default -> "";
+        };
         return result;
     }
 
diff --git a/jabel-javac-plugin/src/main/java/com/github/bsideup/jabel/JabelCompilerPlugin.java b/jabel-javac-plugin/src/main/java/com/github/bsideup/jabel/JabelCompilerPlugin.java
index 0a5aeb1..aa0a15d 100644
--- a/jabel-javac-plugin/src/main/java/com/github/bsideup/jabel/JabelCompilerPlugin.java
+++ b/jabel-javac-plugin/src/main/java/com/github/bsideup/jabel/JabelCompilerPlugin.java
@@ -157,6 +157,7 @@ public class JabelCompilerPlugin implements Plugin {
                 case "EFFECTIVELY_FINAL_VARIABLES_IN_TRY_WITH_RESOURCES":
                 case "TEXT_BLOCKS":
                 case "PATTERN_MATCHING_IN_INSTANCEOF":
+                case "PATTERN_SWITCH":
                 case "REIFIABLE_TYPES_INSTANCEOF":
                 case "RECORDS":
                     //noinspection UnusedAssignment
```

Running `./gradlew test` this gives:
```
An exception has occurred in the compiler (21.0.1). Please file a bug against the Java compiler via the Java bug reporting page (https://bugreport.java.com) after checking the Bug Database (https://bugs.java.com) for duplicates. Include your program, the following diagnostic, and the parameters passed to the Java compiler in your report. Thank you.
com.sun.tools.javac.code.Symbol$CompletionFailure: class file for java.lang.runtime.SwitchBootstraps not found
```

The error is awkward, but I validated that if one targets JDK 21, then the `.class` file will contain a reference to [java.lang.runtime.SwitchBootstraps](https://github.com/openjdk/jdk/blob/437342b93e9e66340ac57bd1c6fdc948b3302db0/src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java), which means that indeed won't work with JDK 11 (as there that class isn't available).